### PR TITLE
Hooks doc updates

### PR DESCRIPTION
**Pull request by hwillson**
This PR will track all Apollo Client doc updates to cover the new React Apollo Hooks functionality. We're adjusting the docs to be Hooks first, but will keep render prop and HOC code samples available via a code sample type toggle, like:

Screenshot 2019-08-02 07 37 15

### CODE REVIEW

**Comment by hwillson** [comment]
I'm going to take this out of draft since the bulk of the new Hooks docs work has been completed. There are still a few things to complete (which I'll list below), but they shouldn't stand in the way of review.

Remaining todos:

- Update the Pupstagram reference app to use Hooks (and get rid of `apollo-link-state` while we're at it)
- Add `Hooks (TypeScript)` and `Render props (TypeScript)` code blocks to some of the main code samples
- Add a brief note or section somwhere that explains why the bulk of the docs content has been updated to promote Hooks first, and how render prop / HOC code samples can still be referenced
- If Gatsby doesn't merge/release https://github.com/gatsbyjs/gatsby/pull/16344 on Monday, go through each `.mdx` file and convert all `###`'s to `<h3 />`'s

**Comment by peggyrayzis** [comment]
Let's do TypeScript as a fast follow up. We can promote it with a new TypeScript getting started guide to be released in a week or two :)

Just made lots of edits. I think it's ready to go for launch day. There's definitely some cleaning up I want to tackle in the future (mostly moving many of the more detailed examples that exist in the HOC docs to the hooks docs), but I don't think it needs to block the launch.

Thank you for all the hard work @hwillson !! 🚀

**Comment by hwillson** [comment]
The changes look great @peggyrayzis, and sounds good about TS - thanks very much!

File: docs/source/features/pagination.mdx [hunk 1]
```diff
@@ -95,15 +205,15 @@ const MoreCommentsQuery = gql`
 `;
 
 const CommentsWithData = () => (
-  <Query query={CommentsQuery}>
+  <Query query={MORE_COMMENTS_QUERY}>
```
Comment (liuderchi) [inline docs/source/features/pagination.mdx]: @hwillson I found a typo here.
Perhaps this should be renamed as another different query `COMMENTS_QUERY` ?
(c.f. MoreCommentsQuery -> `MORE_COMMENTS_QUERY`; CommentsQuery -> `COMMENTS_QUERY`)

Also, [L156 has similar issue](https://github.com/apollographql/apollo-client/pull/5119/files#diff-bbc1ce0c75732db74d56ef2b937a48efR156).
Do I need to file a new issue addressing it?

Comment (liuderchi) [reply docs/source/features/pagination.mdx]: cc @peggyrayzis

**Pull request merged**